packages/xod-project/src/internal/patchPathUtils.js

```diff
@@ -1,14 +1,20 @@
 import * as R from 'ramda';
+import { notEmpty } from 'xod-func-tools';
 import * as CONST from '../constants';
 
-// :: String -> Boolean
-export const isValidIdentifier = R.allPass([
+export const isProjectNameValid = R.allPass([
   R.test(/[a-z0-9-]*/), // only lowercase alphanumeric and hypen
   R.complement(R.test(/^-/)), // can't start with hypen
   R.complement(R.test(/-$/)), // can't end with hypen
   R.complement(R.test(/--/)), // only one hypen in row
 ]);
 
+// :: String -> Boolean
+export const isValidIdentifier = R.both(
+  isProjectNameValid,
+  notEmpty
+);
```

Comment:
:man_facepalming: Do general identifier rules depend on project naming rules?

Could it be:

```js
export const isValidIdentifier = R.allPass([
  R.test(/[a-z0-9-]+/), // <-- plus here
  ...
]);

export const isValidProjectName = R.either(
  isValidIdentifier,
  R.isEmpty
);
```